### PR TITLE
C17 changes try again

### PR DESCRIPTION
**Pull request by graydon**
This attempts to merge the C++17 changes again, this time with a somewhat horribe (but I think safe) workaround for the integration test that failed previously.

### CODE REVIEW

File: src/herder/test/UpgradesTests.cpp [hunk 1]
```diff
@@ -2319,3 +2319,33 @@ TEST_CASE("validate upgrade expiration logic", "[upgrades]")
         REQUIRE(upgrades.mBaseReserve);
     }
 }
+
+TEST_CASE("upgrade from cpp14 serialized data", "[upgrades]")
+{
+    std::string in = "{\n"
```
Comment (MonsieurNicolas) [inline src/herder/test/UpgradesTests.cpp:2325]: please use a raw string litteral instead of escaping like this

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]
minor suggestion to make it easier to maintain

File: src/herder/test/UpgradesTests.cpp (hunk 1 above)
Comment (graydon) [reply src/herder/test/UpgradesTests.cpp]: fixed

**Comment by MonsieurNicolas** [comment]
r+ fcf6fe7df770ae402a910a62ff5ed05f617f0c88

File: src/herder/test/UpgradesTests.cpp (hunk 1 above)
Comment (joko300370) [reply src/herder/test/UpgradesTests.cpp]: joko300370 move and run

**Pull request merged**